src/Datadog.Trace.ClrProfiler.Managed/Emit/ObjectExtensions.cs

```diff
@@ -271,5 +271,40 @@ private static string GetKey<TResult>(string name, Type type)
             dynamicMethod.Return();
             return dynamicMethod.CreateDelegate();
         }
+
+        private readonly struct PropertyFetcherCacheKey : IEquatable<PropertyFetcherCacheKey>
+        {
+            public readonly Type Type1;
+            public readonly Type Type2;
+            public readonly string Name;
+
+            public PropertyFetcherCacheKey(Type type1, Type type2, string name)
+            {
+                Type1 = type1;
+                Type2 = type2;
+                Name = name;
+            }
+
+            public bool Equals(PropertyFetcherCacheKey other)
+            {
+                return Equals(Type1, other.Type1) && Equals(Type2, other.Type2) && Name == other.Name;
+            }
+
+            public override bool Equals(object obj)
+            {
+                return obj is PropertyFetcherCacheKey other && Equals(other);
+            }
+
+            public override int GetHashCode()
+            {
+                unchecked
+                {
+                    var hashCode = (Type1 != null ? Type1.GetHashCode() : 0);
+                    hashCode = (hashCode * 397) ^ (Type2 != null ? Type2.GetHashCode() : 0);
+                    hashCode = (hashCode * 397) ^ (Name != null ? Name.GetHashCode() : 0);
```

Comment:
Resharper has an option when creating these methods that skips the null checks if the fields can never be null.

Reply:
Fixed